assets/src/components/PostMeta/withPostMeta.js

```diff
@@ -14,13 +14,38 @@ const getValuePropName = ( Component ) => {
 
 const { apiFetch } = wp;
 
+const getValueFromProps = ( props ) => {
+  const { metaKey, postMeta } = props;
+
+  const metaValue = postMeta[ metaKey ];
+
+  let value;
+  // Use the default value if what is stored in the post meta is empty, or if it isn't one of the listed options.
+  if (
+    props.defaultValue
+    && (
+      !metaValue
+      || !props.options.some( option => option.value === metaValue )
+    )
+  ) {
+    value = props.defaultValue;
+  } else {
+    value = metaValue;
+  }
+
+  return value;
+}
+
 export function withPostMeta( WrappedComponent ) {
 
   class WrappingComponent extends Component {
     constructor( props ) {
       super( props );
       this.handleChange = this.handleChange.bind( this );
       this.valuePropName = getValuePropName( WrappedComponent );
+      this.state = {
+        value: getValueFromProps( props )
```

Comment:
Hey @Inwerpsel I think your previous comment here got lost, I updated this to `getValueFromProps` but I'm keeping this in the constructor since otherwise it would trigger a warning: `react-dom.js:539 Warning: `WrappingComponent` uses `getDerivedStateFromProps` but its initial state is undefined. This is not recommended. Instead, define the initial state by assigning an object to `this.state` in the constructor of `WrappingComponent`. This ensures that `getDerivedStateFromProps` arguments have a consistent shape.`

Reply:
You could still assign `null` to `value` so that `getValueFromProps` isn't executed twice?

Reply:
This got outdated too.